### ISSUE TITLE
[BUG] Kill-Process GUI shown when nodemon restarts

### ISSUE THREAD

**Issue by Nigelrex** [remy/nodemon#1879]
- `nodemon -v`: v2.0.9
- `node -v`: v12.9.1
- Operating system/terminal environment: Windows 7, x64 / CMD
- Using Docker? What image: NONE
- Command you ran: nodemon / nodemon bot.js

### Expected behaviour
During file changes and restarting of nodemon, only the terminal should be showing restarting and the kill-process should be triggered, and there should be no-GUI popping up with the name Kill-process.

### Actual behaviour
Same as above but the kill-process Terminal pops up during file changes and restart

### Steps to reproduce
I am not sure how that came, I started nodemon as usual and this happened

---

If applicable, please append the `--dump` flag on your command and include the output here **ensuring to remove any sensitive/personal details or tokens**.

```console
--------------
node: v12.9.1
nodemon: 2.0.9
command: C:\Program Files\nodejs\node.exe C:\Users\DELL\AppData\Roaming\npm\node_modules\nodemon\bin\nodemon.js --dump
cwd: D:\Discord\Discord Bots\MY Discord Bots\Laterex
OS: win32 x64
--------------
{
  run: false,
  system: { cwd: 'D:\\Discord\\Discord Bots\\MY Discord Bots\\Laterex' },
  required: false,
  dirs: [ 'D:\\Discord\\Discord Bots\\MY Discord Bots\\Laterex' ],
  timeout: 1000,
  options: {
    dump: true,
    ignore: [
      '**/.git/**',
      '**/.nyc_output/**',
      '**/.sass-cache/**',
      '**/bower_components/**',
      '**/coverage/**',
      '**/node_modules/**',
      re: /.*.*\/\.git\/.*.*|.*.*\/\.nyc_output\/.*.*|.*.*\/\.sass\-cache\/.*.*|.*.*\/bower_components\/.*.*|.*.*\/coverage\/.*.*|.*.*\/node_modules\/.*.*/
    ],
    watch: [ '*.*', re: /.*\..*/ ],
    monitor: [
      '*.*',
      '!**/.git/**',
      '!**/.nyc_output/**',
      '!**/.sass-cache/**',
      '!**/bower_components/**',
      '!**/coverage/**',
      '!**/node_modules/**'
    ],
    ignoreRoot: [
      '**/.git/**',
      '**/.nyc_output/**',
      '**/.sass-cache/**',
      '**/bower_components/**',
      '**/coverage/**',
      '**/node_modules/**'
    ],
    restartable: 'rs',
    colours: true,
    execMap: { py: 'python', rb: 'ruby', ts: 'ts-node' },
    stdin: true,
    runOnChangeOnly: false,
    verbose: false,
    signal: 'SIGUSR2',
    stdout: true,
    watchOptions: {},
    execOptions: {
      script: 'bot.js',
      exec: 'node',
      args: [],
      scriptPosition: 0,
      nodeArgs: undefined,
      execArgs: [],
      ext: 'js,mjs,json',
      env: {}
    }
  },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [],
  watchInterval: null,
  signal: 'SIGUSR2',
  command: {
    raw: { executable: 'node', args: [ 'bot.js' ] },
    string: 'node bot.js'
  }
}
--------------
```

**Comment by remy** [comment]
What GUI are you seeing? If you can provide a video or at least screenshots it would be useful.

Once you've done that, can you also test by installing `npm i -g nodemon@debug` and repeat the test?

**Comment by Nigelrex** [comment]
ok, like once the GUI freezed and i took a screenshot and started to write this bug, but soon my PC crashed and i lost the screen shot

**Comment by Nigelrex** [comment]
![NODEMON BUG](https://user-images.githubusercontent.com/75080457/124700394-4e39a600-df0a-11eb-90e4-853f11b93c3d.gif)

@remy
Here is the proof of what is happening

**Comment by Nigelrex** [comment]
And the same happens with `npm i -g nodemon@debug`

**Comment by Nigelrex** [comment]
Also the Bug #1876 and #1872 looks similar to me

**Comment by remy** [comment]
I hadn't spotted you're on windows 7. Can you try with nodemon@2.0.7 and let me know if restarting works as expected?

**Comment by Nigelrex** [comment]
I was acctually using Nodemon@2.0.6 and then this happened, then I updated to see if it still remains a problem, and it was

**Comment by Nigelrex** [comment]
Oh wait I just tested @2.0.7 it worked as expected, also can you please put this issue up somewhere noticeable so if people have the same problem like me they can view this issue,

Thanks you. :pray:

**Comment by Nigelrex** [comment]
Instead of me closing it, you check everything and close it yourself, or you will be searching this issue later!

**Comment by remy** [comment]
What I'll *try* to do is fork the logic handling for the windows restart process the use the windows-kill.exe only if it's above or including windows 10. This way it won't catch people out.

**Comment by Nigelrex** [comment]
Nice so no more windows-kill.exe right, then how will we stop the nodemon process. eg: `process.exit();` is stopping the node,

For nodemon it's 
```js
process.once("SIGUSR2", function () {
            gracefulShutdown(function () {
              process.off();
              process.kill(process.pid, "SIGUSR2");
            });```

then how will we stop the nodemon process *if* you implement it?

**Comment by remy** [comment]
I'll restore the fork of code logic that handles the restart in 2.0.7 for < windows 10.

**Comment by Nigelrex** [comment]
oK cool, I think you are close this issue

**Comment by remy** [comment]
Can you try to install (an updated) `npm i -g nodemon@debug` and retest? I've added forked logic for Windows < 10.

**Comment by DineshRout779** [comment]
I am having the same issue in my nodejs project. As soon as I update the app.js in my project, error shows as 
" Command failed : wmic error."  and it is killing the process.
Detailed error:- https://stackoverflow.com/questions/68250987/command-failed-wmic-process-after-updating-npm-in-express-app/68270072#68270072

**Comment by remy** [comment]
@DineshRout779 you'll need to actually tell us what version of nodemon you're using…

**Comment by remy** [comment]
@Nigelrex this landed in 2.0.10 - so windows-kill is only used in Windows 10 and above.

**Comment by Nigelrex** [comment]
> @Nigelrex this landed in 2.0.10 - so windows-kill is only used in Windows 10 and above.

but it worked for me :smile:

**Comment by DineshRout779** [comment]
> @DineshRout779 you'll need to actually tell us what version of nodemon you're using…

Nodemon :- 2.0.10
Windows :- 7 32 bit
